src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -217,23 +217,41 @@ public function regenerate($destroy = false, $lifetime = null)
      */
     public function save()
     {
+        $session = $_SESSION;
+
+        foreach ($this->bags as $bag) {
+            if (empty($_SESSION[$key = $bag->getStorageKey()])) {
+                unset($_SESSION[$key]);
+            }
+        }
+        if (array($key = $this->metadataBag->getStorageKey()) === array_keys($_SESSION)) {
+            unset($_SESSION[$key]);
+        }
```

Comment:
If the session is empty at this point, it will be an empty array, right? Will that serialize to an empty string? Or will it be something like `a:0:{}`?

I am a little unfamiliar with how your tests work, could you please be so kind to point out for me the file and line where you test this?

Reply:
Correct: sessions serialization is special and doesn't use "serialiaze()" but a variant of it. And the empty session array is represented by the empty string (that's why session handlers return the empty string when they have no data for a specific session id.)

Reply:
Just out of curiosity, will `null` also serialize to an empty string?

You did not tell me about the test, I assume the full use of a nativeSessionStorage bag is covered in the tests?

Reply:
"null" is not a valid value for `$_SESSION` so I don't know how PHP will behave if you such such a thing?

Reply:
Oh, about the tests, there is none right now, I'm on it :)

Reply:
I just read up a bit on this, have a look at http://php.net/manual/en/function.session-decode.php and http://php.net/manual/en/session.configuration.php#ini.session.serialize-handler

It seems like this will only be working when  `session.serialize_handler` is set to the default `php` one?

Reply:
I don't know how the other serialize handler serialize the empty array. Let's change the check and use `empty($_SESSION)` instead.

Reply:
Since you need to do that in the `write` function of the session handler, you may want to use this https://packagist.org/packages/wikimedia/php-session-serializer

Reply:
Ok, scrap that, you have also things like [session.serialize_handler=igbinary](https://github.com/igbinary/igbinary) so the wikimedia/php-session-serializer would not cover all use cases.

Reply:
I opened https://github.com/igbinary/igbinary/issues/146 about igbinary, and meanwhile it is now handled in the code. All other serialization handler store the empty session as the empty string.

Reply:
Now tested, see https://github.com/symfony/symfony/pull/24523/files#diff-df2c28eb378a2dc581bb6b53fff80661

Reply:
I do not see where you have tested with all the different session handlers.

And in which test do you test that it does not start a session if just just read from a session bag?

Reply:
The existing session handlers are already tested, there's no need to test them again. See fixture I linked.

Reply:
storage.php